packages/opentelemetry-resources/test/detectors/ProcessDetector.test.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { processDetector, Resource } from '../../src';
+import {
+  assertProcessResource,
+  assertEmptyResource,
+} from '../util/resource-assertions';
+import { NoopLogger } from '@opentelemetry/core';
+
+describe('processDetector()', () => {
+  it('should return resource information from process', async () => {
+    Object.defineProperty(process, 'pid', {
```

Comment:
please use sinon sandbox for mocking the property on process

```
  let sandbox: sinon.SinonSandbox;

  beforeEach(() => {
    sandbox = sinon.createSandbox();
  });

  afterEach(() => {
    sandbox.restore();
  });

  // and then your tests
  it('should ...', ()=> {
    sandbox.stub(process, 'pid').value(1234);
    // your test goes here
  });
```

Reply:
Thanks TIL. I've updated the test cases.